examples/parallel_hashmap_main.cc

```diff
@@ -0,0 +1,21 @@
+#include <iostream>
+
+#include "parallel_hashmap/phmap.h"
+
+// Example Parallel Hashmap usage
+// https://github.com/greg7mdp/parallel-hashmap
```

Comment:
Just want to clarify one thing here - the default option for phmap does not have internal locking so users need to synchronize read and write explicitly. 

phmap can be configured to be inherently thread-safe (e.g. read / write without having to worry about race condition) by taking a lock type (e.g. absl::Mutex( as its latest template variable. Example for this can be found at :

https://github.com/greg7mdp/parallel-hashmap/blob/master/examples/bench.cc

More specifically see lin 63 - 70 from the above file. 

I think this is what we need, because otherwise we still need to manage a handful of locks ourselves. 

It is probably worth defining a customized alias in util, such as the ones used in the bench.cc above, so that one we define a parallel flat hashmap, we have internal locks for it. The reason why I propose an alias instead of defining everything explicitly is that incorporating the template variables makes the definition extremely verbose, for example, the extra variables for defining the map look like:

        #define EXTRAARGS , NMSP::container_internal::hash_default_hash<K>, \
                            NMSP::container_internal::hash_default_eq<K>, \
                            std::allocator<std::pair<const K, V>>, 4, NMSP::NullMutex

Simplifying the above would be helpful, and I think it is worth including an example in this category.

Reply:
Can you just push a commit to this branch for the example?

Reply:
Yes, let me do so